yt_dlp/extractor/tencent.py

```diff
@@ -97,21 +130,36 @@ def _extract_video_native_subtitles(self, api_response, subtitles_format):
 
         return subtitles
 
-    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
-        formats, subtitles = [], {}
-        for video_format, subtitle_format, video_quality in (
-                # '': 480p, 'shd': 720p, 'fhd': 1080p
-                ('mp4', 'srt', ''), ('hls', 'vtt', 'shd'), ('hls', 'vtt', 'fhd')):
-            api_response = self._get_video_api_response(
-                url, video_id, series_id, subtitle_format, video_format, video_quality)
+    def _extract_all_video_quality(self, api_response):
+        quality = []
+        formats_response = api_response['fl']['fi']
+        for f in formats_response:
+            quality.append(f.get('name'))
+        return quality
 
-            if api_response.get('em') != 0 and api_response.get('exem') != 0:
-                if '您所在区域暂无此内容版权' in api_response.get('msg'):
-                    self.raise_geo_restricted()
-                raise ExtractorError(f'Tencent said: {api_response.get("msg")}')
+    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
+        first_api_response = self._get_video_api_response(url, video_id, series_id, 'srt', 'hls', 'hd')
+        self._check_api_response(first_api_response)
+        api_responses = [{
+            'response': first_api_response,
+            'subtitle_format': 'srt',
+        }]
```

Comment:
I know you didn't modify _extract_video_native_subtitles but I think we can avoid passing 'subtitle_format' if we check its 'captionType'. 

I think it would simplify this, using only a list of api_responses 

```python
    def _extract_video_native_subtitles(self, api_response):
        subtitles = {}
        for subtitle in traverse_obj(api_response, ('sfl', 'fi')) or ():
            subtitles.setdefault(subtitle['lang'].lower(), []).append({
                'url': subtitle['url'],
                'ext': 'vtt' if subtitle.get('captionType') == '1' else 'srt',
                'protocol': 'm3u8_native' if determine_ext(subtitle['url']) == 'm3u8' else 'http',
            })

        return subtitles
```

Reply:
Ok, I'll look at what the API returns later and improve it.